### PR TITLE
Move interfaces/p2p to a new crate net/p2p

### LINKED ISSUE #7598: extract interfaces/p2p to standalone crate under net/p2p
### Describe the feature

we want to get rid of the interfaces crate, all p2p logic should be moved to a standalone crate under net/

https://github.com/paradigmxyz/reth/tree/dd28a4ca730f5d213861ab23ca012896c1d19c34/crates/interfaces/src/p2p

should also be moved there:

https://github.com/paradigmxyz/reth/blob/dd28a4ca730f5d213861ab23ca012896c1d19c34/crates/interfaces/src/sync.rs#L8-L8

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by supernovahs**
closes #7598

### CODE REVIEW

**Comment by supernovahs** [comment]
Figuring this out then clean up 
```
warning: `reth-rpc-builder` (lib) generated 2 warnings
error[E0599]: the method `value_parser` exists for reference `&&&&&&_AutoValueParser<LogLevel>`, but its trait bounds were not satisfied
    --> crates/node-core/src/args/database_args.rs:12:5
     |
12   |     /// Database logging level. Levels higher than "notice" require a debug build.
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method cannot be called on `&&&&&&_AutoValueParser<LogLevel>` due to unsatisfied trait bounds
     |
    ::: /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.5.2/src/builder/value_parser.rs:2462:1
     |
2462 | pub struct _AutoValueParser<T>(std::marker::PhantomData<T>);
     | ------------------------------ doesn't satisfy `_: _ValueParserViaParse`
     |
    ::: /home/runner/work/reth/reth/crates/net/p2p/src/db.rs:104:1
     |
104  | pub enum LogLevel {
     | ----------------- doesn't satisfy 7 bounds
     |
     = note: the following trait bounds were not satisfied:
             `LogLevel: ValueEnum`
             which is required by `&&&&&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaValueEnum`
             `LogLevel: ValueParserFactory`
             which is required by `&&&&&&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaFactory`
             `LogLevel: From<OsString>`
             which is required by `&&&&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaFromOsString`
             `LogLevel: From<&'s std::ffi::OsStr>`
             which is required by `&&&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaFromOsStr`
             `LogLevel: From<std::string::String>`
             which is required by `&&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaFromString`
             `LogLevel: From<&'s str>`
             which is required by `&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaFromStr`
             `LogLevel: FromStr`
             which is required by `_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaParse`
     = note: this error originates in the macro `clap::value_parser` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0599`.
error: could not compile `reth-node-core` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error[E0599]: the method `value_parser` exists for reference `&&&&&&_AutoValueParser<LogLevel>`, but its trait bounds were not satisfied
    --> crates/node-core/src/args/database_args.rs:12:5
     |
12   |     /// Database logging level. Levels higher than "notice" require a debug build.
     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ method cannot be called on `&&&&&&_AutoValueParser<LogLevel>` due to unsatisfied trait bounds
     |
    ::: /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clap_builder-4.5.2/src/builder/value_parser.rs:[246](https://github.com/paradigmxyz/reth/actions/runs/8770603685/job/24067351439?pr=7779#step:6:247)2:1
     |
2462 | pub struct _AutoValueParser<T>(std::marker::PhantomData<T>);
     | ------------------------------ doesn't satisfy `_: _ValueParserViaParse`
     |
    ::: /home/runner/work/reth/reth/crates/net/p2p/src/db.rs:104:1
     |
104  | pub enum LogLevel {
     | ----------------- doesn't satisfy 7 bounds
     |
     = note: the following trait bounds were not satisfied:
             `LogLevel: ValueEnum`
             which is required by `&&&&&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaValueEnum`
             `LogLevel: ValueParserFactory`
             which is required by `&&&&&&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaFactory`
             `LogLevel: From<OsString>`
             which is required by `&&&&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaFromOsString`
             `LogLevel: From<&'s std::ffi::OsStr>`
             which is required by `&&&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaFromOsStr`
             `LogLevel: From<std::string::String>`
             which is required by `&&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaFromString`
             `LogLevel: From<&'s str>`
             which is required by `&_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaFromStr`
             `LogLevel: std::str::FromStr`
             which is required by `_AutoValueParser<LogLevel>: clap::builder::via_prelude::_ValueParserViaParse`
     = note: this error originates in the macro `clap::value_parser` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `reth-node-core` (lib test) due to 2 previous errors
error: command `/home/runner/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo test --no-run --message-format json-render-diagnostics --workspace --exclude examples --exclude ef-tests --features 'asm-keccak optimism' --locked` exited with code 101
Error: Process completed with exit code 101.
```

**Comment by supernovahs** [comment]
Let me know for refactors needed 

- crate name `reth-net-p2p`

- I shifted the `consensus` ,`sync`,` provider` , `db` files and `test-utils` from the interfaces to p2p as those were needed.
- Also, now directly importing `ForkchoiceState` from alloy rpc types engine directly instead of using reth rpc types

**Comment by supernovahs** [comment]
just saw #7757 .will wait forit

File: crates/consensus/beacon-core/src/lib.rs [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
 
 use reth_consensus_common::validation;
-use reth_interfaces::consensus::{Consensus, ConsensusError};
+use reth_net_p2p::consensus::{Consensus, ConsensusError};
```
Comment (mattsse) [inline crates/consensus/beacon-core/src/lib.rs:12]: this should not be imported from the p2p crate

File: crates/net/p2p/src/lib.rs
Comment (mattsse) [inline crates/net/p2p/src/lib.rs:1]: this needs the regular lib.rs header, see other crates

**Review by mattsse — CHANGES_REQUESTED** [review]
few nits, then this is blocked by the consensus crate pr

**Comment by mattsse** [comment]
@supernovahs this is unblocked now

**Comment by supernovahs** [comment]
et me know if I missed something

**Comment by supernovahs** [comment]
It would probably difficult to review now for you. Should have started with fresh main

**Comment by emhane** [comment]
will you start again here on same branch @supernovahs ?

**Comment by supernovahs** [comment]
Would start on a new branch now probably

**Comment by mattsse** [comment]
yeah this drifted too much, so best to redo before wasting time resolving conflicts imo

**Pull request closed without merging**